[vtk] Remove a spurious dependency on VTK::utf8

This patch has been upstreamed and can likely be dropped in a future
update to Drake's VTK.

See https://gitlab.kitware.com/vtk/vtk/-/merge_requests/10793.

--- Common/Core/vtk.module
+++ Common/Core/vtk.module
@@ -11,8 +11,6 @@ GROUPS
 DEPENDS
   VTK::kwiml
   VTK::vtksys
-PRIVATE_DEPENDS
-  VTK::utf8
 OPTIONAL_DEPENDS
   VTK::loguru
 TEST_DEPENDS
